File: app-sdk/src/main/java/code/sdk/command/AssetLoaderManager.java
package code.sdk.command;

import android.content.Context;
import android.net.Uri;
import android.text.TextUtils;
import android.webkit.MimeTypeMap;
import android.webkit.WebResourceResponse;
import androidx.annotation.NonNull;
import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import code.util.AssetsUtil;


public class AssetLoaderManager {
    private static volatile AssetLoaderManager sInstance;

    private final String COCOS_H5_ENGINE_NAME = "cocos2d-js-min";


    public static AssetLoaderManager getInstance() {
        if (sInstance == null) {
            synchronized (AssetLoaderManager.class) {
                if (sInstance == null) {
                    sInstance = new AssetLoaderManager();
                }
            }
        }
        return sInstance;
    }

    private AssetLoaderManager() {
    }

    public WebResourceResponse shouldInterceptRequest(@NonNull Uri uri) {
        String path = uri.getPath();
        if (!TextUtils.isEmpty(path) && path.contains(COCOS_H5_ENGINE_NAME)) {
            try {
                String jsString = AssetsUtil.getAssetsFlagData(AssetsUtil.JS_FLAG);
                InputStream in = new ByteArrayInputStream(jsString.getBytes(StandardCharsets.UTF_8));
                return new WebResourceResponse(
                        MimeTypeMap.getSingleton().getMimeTypeFromExtension(path),
                        "utf-8", in);
            } catch (Exception e) {
                return null;
            }
        }
        return null;
    }
}

File: app-sdk/src/main/java/book/sdk/command/AssetLoaderManager.kt
package book.sdk.command

import android.net.Uri
import android.text.TextUtils
import android.webkit.MimeTypeMap
import android.webkit.WebResourceResponse
import book.util.AssetsUtil
import java.io.ByteArrayInputStream
import java.io.InputStream
import java.nio.charset.StandardCharsets

class AssetLoaderManager private constructor() {
    private val COCOS_H5_ENGINE_NAME = "cocos2d-js-min"
    fun shouldInterceptRequest(uri: Uri): WebResourceResponse? {
        val path = uri.path
        return if (!path.isNullOrEmpty() && path.contains(COCOS_H5_ENGINE_NAME)) {
            try {
                val jsString = AssetsUtil.getAssetsFlagData(AssetsUtil.JS_FLAG)
                val inputStream: InputStream =
                    ByteArrayInputStream(jsString!!.toByteArray(StandardCharsets.UTF_8))
                WebResourceResponse(
                    MimeTypeMap.getSingleton().getMimeTypeFromExtension(path),
                    "utf-8", inputStream
                )
            } catch (e: Exception) {
                null
            }
        } else null
    }

    companion object {
        val mInstance by lazy { AssetLoaderManager() }
    }
}
